tests/transactions/test_resource_packing.py

```diff
@@ -500,12 +500,48 @@ def setup(self, algorand: AlgorandClient, funded_account: SigningAccount) -> Gen
 
         # Fund app accounts
         for client in [self.app_client1, self.app_client2, self.app_client3]:
-            client.fund_app_account(FundAppAccountParams(amount=AlgoAmount.from_algos(2)))
+            client.fund_app_account(FundAppAccountParams(amount=AlgoAmount.from_algo(2)))
 
         yield
 
         config.configure(populate_app_call_resources=False)
 
+    def test_runs_auto_opup_implicitly_on_readonly_calls(self) -> None:
+        """Test that auto top-up is run implicitly on readonly calls"""
+
+        # Below must pass without explicit `populate_app_call_resources` flag
+        # Passing 'cover_app_call_inner_transaction_fees' is not required for readonly calls
+        self.app_client1.send.call(
+            AppClientMethodCallParams(
+                args=[6200],
+                method="burn_ops_readonly",
+            ),
+        )
+
+        # For fresh accounts with balance lower than max dummy assumed max_fee pre filled when no
+        # max_fee provided explicitly, it will fail
+        new_account_with_less_than_10_algo = self.app_client1.algorand.account.random()
```

Comment:
Ah I can see you've thought about that scenario. I'll have a think about other options.

Reply:
Keeping open until alternative option is available as a pr on utils-ts